scapy/fields.py

```diff
@@ -1230,26 +1230,23 @@ def i2repr(self, pkt, x):
         return bytes_hex(x).decode()
 
 
-class XStrLenField(StrLenField):
-    """
-    StrLenField which value is printed as hexadecimal.
-    """
-
+class _XStrField:
     def i2repr(self, pkt, x):
         if not x:
             return repr(x)
         return bytes_hex(x[:self.length_from(pkt)]).decode()
 
 
-class XStrFixedLenField(StrFixedLenField):
+class XStrLenField(StrLenField, XStrField):
```

Comment:
Note to self: that should actually be _XStrField.
Maybe rename the second field for clarity ?